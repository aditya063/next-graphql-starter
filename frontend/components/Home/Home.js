// import { IdeaCardContainer } from '..';
import IdeaCardContainer from '../IdeaCardContainer/IdeaCardContainer';
import * as sc from './Home.style';

const Home = React.memo(() => (
  <sc.main>
    <IdeaCardContainer />
  </sc.main>
));

export default Home;
