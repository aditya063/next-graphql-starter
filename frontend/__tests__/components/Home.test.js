import React from 'react';
import { shallow } from 'enzyme';
import Home from '../../components/Home/Home';

describe('Home', () => {
  let mockProps;
  let wrapper;

  beforeEach(() => {
    mockProps = {};
    wrapper = shallow(<Home {...mockProps} />);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
