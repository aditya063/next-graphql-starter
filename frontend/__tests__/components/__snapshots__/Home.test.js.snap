// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches snapshot - loading 1`] = `
Array [
  .c0 {
  grid-area: main;
}

<Homestyle__main
    snapshot="Home"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "Homestyle__main-sc-8q62fs-0",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "grid-area:main;",
            ],
          },
          "displayName": "Homestyle__main",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "Homestyle__main-sc-8q62fs-0",
          "target": "main",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      snapshot="Home"
    >
      <main
        className="c0"
      >
        <Component>
          <Query
            errorPolicy="all"
            query={
              Object {
                "definitions": Array [
                  Object {
                    "directives": Array [],
                    "kind": "OperationDefinition",
                    "name": Object {
                      "kind": "Name",
                      "value": "ME_QUERY",
                    },
                    "operation": "query",
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "me",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "id",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "email",
                                },
                                "selectionSet": undefined,
                              },
                            ],
                          },
                        },
                      ],
                    },
                    "variableDefinitions": Array [],
                  },
                ],
                "kind": "Document",
                "loc": Object {
                  "end": 60,
                  "start": 0,
                },
              }
            }
          >
            <Component>
              <p
                snapshot="DisplayLoading"
              >
                Loading ...
              </p>
            </Component>
          </Query>
        </Component>
      </main>
    </StyledComponent>
  </Homestyle__main>,
  .c0 {
  grid-area: main;
}

<StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "Homestyle__main-sc-8q62fs-0",
          "isStatic": true,
          "lastClassName": "c0",
          "rules": Array [
            "grid-area:main;",
          ],
        },
        "displayName": "Homestyle__main",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "Homestyle__main-sc-8q62fs-0",
        "target": "main",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    snapshot="Home"
  >
    <main
      className="c0"
    >
      <Component>
        <Query
          errorPolicy="all"
          query={
            Object {
              "definitions": Array [
                Object {
                  "directives": Array [],
                  "kind": "OperationDefinition",
                  "name": Object {
                    "kind": "Name",
                    "value": "ME_QUERY",
                  },
                  "operation": "query",
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "me",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "id",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "email",
                              },
                              "selectionSet": undefined,
                            },
                          ],
                        },
                      },
                    ],
                  },
                  "variableDefinitions": Array [],
                },
              ],
              "kind": "Document",
              "loc": Object {
                "end": 60,
                "start": 0,
              },
            }
          }
        >
          <Component>
            <p
              snapshot="DisplayLoading"
            >
              Loading ...
            </p>
          </Component>
        </Query>
      </Component>
    </main>
  </StyledComponent>,
]
`;
